pkg/controller/v1alpha1/inferencegraph/knative_reconciler.go

```diff
@@ -90,41 +90,41 @@ func reconcileKsvc(desired *knservingv1.Service, existing *knservingv1.Service)
 func (r *GraphKnativeServiceReconciler) Reconcile() (*knservingv1.ServiceStatus, error) {
 	desired := r.Service
 	existing := &knservingv1.Service{}
-	err := r.client.Get(context.TODO(), types.NamespacedName{Name: desired.Name, Namespace: desired.Namespace}, existing)
-	if err != nil {
-		if apierr.IsNotFound(err) {
-			log.Info("Creating inference graph knative service", "namespace", desired.Namespace, "name", desired.Name)
-			return &desired.Status, r.client.Create(context.TODO(), desired)
-		}
-		return nil, err
-	}
-
-	// Set ResourceVersion which is required for update operation.
-	desired.ResourceVersion = existing.ResourceVersion
-	// Add immutable annotations to avoid validation error during dry-run update.
-	desired.Annotations[knserving.CreatorAnnotation] = existing.Annotations[knserving.CreatorAnnotation]
-	desired.Annotations[knserving.UpdaterAnnotation] = existing.Annotations[knserving.UpdaterAnnotation]
-
-	// Do a dry-run update to avoid diffs generated because knative defaulter webhook is
-	// called when creating or updating the knative service. This will populate our local knative service object with any default values
-	// that are present on the remote version.
-	if err = r.client.Update(context.TODO(), desired, client.DryRunAll); err != nil {
-		log.Error(err, "Failed to perform dry-run create of knative service", "service", desired.Name)
-		return nil, err
-	}
 
-	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
it is better to wrap `retry.RetryOnConflict` on `Reconcile` instead if the entire function needs to be in the retry block.